bundle/src/main/java/com/adobe/acs/commons/ccvar/impl/AllPagePropertiesContentVariableProvider.java

```diff
@@ -0,0 +1,100 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+
+import static com.adobe.acs.commons.ccvar.util.PropertyAggregatorUtil.addPropertiesToMap;
+
+@Component(service = ContentVariableProvider.class)
+public class AllPagePropertiesContentVariableProvider implements ContentVariableProvider {
+
+    private static final Logger LOG = LoggerFactory.getLogger(AllPagePropertiesContentVariableProvider.class);
+
+    public static final String PAGE_PROP_PREFIX = "page_properties";
+    private static final String INHERITED_PAGE_PROP_PREFIX = "inherited_page_properties";
+
+    @Reference
+    private PropertyConfigService propertyConfigService;
+
+    @Override
+    public void addProperties(Map<String, Object> map, SlingHttpServletRequest request) {
+        Resource resource = request.getResource();
+
+        PageManager pageManager = resource.getResourceResolver().adaptTo(PageManager.class);
+        if (pageManager == null) {
+            LOG.warn("PageManager was null, skipping properties.");
+            return;
+        }
+        Page page = pageManager.getContainingPage(resource);
+
+        if (page == null) {
+            LOG.warn("No containing page found for resource at {}", resource.getPath());
+            return;
+        }
+
+        // Add current page properties
+        addPagePropertiesToMap(map, page, PAGE_PROP_PREFIX, propertyConfigService);
+
+        // Add inherited page properties
+        while (page != null) {
+            addPagePropertiesToMap(map, page, INHERITED_PAGE_PROP_PREFIX, propertyConfigService);
+            page = page.getParent();
+        }
+    }
+
+    /**
+     * Add the properties of a page to the given map.  Excluded properties are found in the
+     * {@link PropertyConfigService} service.
+     *
+     * @param map                   the map that should be updated with the properties and their values
+     * @param page                  the page containing properties
+     * @param prefix                the prefix to apply to the
+     * @param propertyConfigService the {@link PropertyConfigService} used to check type and exclusion
+     */
+    private void addPagePropertiesToMap(Map<String, Object> map, Page page, String prefix,
+                                              PropertyConfigService propertyConfigService) {
+        ValueMap pageProperties = page.getProperties();
+        addPropertiesToMap(map, pageProperties.entrySet(), prefix, false, propertyConfigService);
+    }
+
+    @Override
+    public boolean accepts(SlingHttpServletRequest request) {
+        PageManager pageManager = request.getResourceResolver().adaptTo(PageManager.class);
+        if (pageManager == null) {
+            LOG.warn("PageManager is null, not accepting this request.");
+            return false;
+        }
+        Page page = pageManager.getContainingPage(request.getResource());
+        return page != null && page.getPath().startsWith("/content");
```

Comment:
should probably be `"/content/"` to ensure it really is a subpage to `/content` and not `/contentishere`

Reply:
updated